pkg/kubectl/cmd/util/shortcut_restmapper.go

```diff
@@ -19,32 +19,32 @@ package util
 import (
 	"strings"
 
+	"github.com/golang/glog"
 	"k8s.io/apimachinery/pkg/api/meta"
 	"k8s.io/apimachinery/pkg/runtime/schema"
 	"k8s.io/client-go/discovery"
 	"k8s.io/kubernetes/pkg/kubectl"
 )
 
-// ShortcutExpander is a RESTMapper that can be used for Kubernetes resources.   It expands the resource first, then invokes the wrapped
-type ShortcutExpander struct {
+// shortcutExpander is a RESTMapper that can be used for Kubernetes resources.   It expands the resource first, then invokes the wrapped
+type shortcutExpander struct {
 	RESTMapper meta.RESTMapper
 
 	All []schema.GroupResource
 
 	discoveryClient discovery.DiscoveryInterface
 }
 
-var _ meta.RESTMapper = &ShortcutExpander{}
+var _ meta.RESTMapper = &shortcutExpander{}
 
-func NewShortcutExpander(delegate meta.RESTMapper, client discovery.DiscoveryInterface) ShortcutExpander {
-	return ShortcutExpander{All: UserResources, RESTMapper: delegate, discoveryClient: client}
-}
-
-func (e ShortcutExpander) getAll() []schema.GroupResource {
-	if e.discoveryClient == nil {
-		return e.All
+func NewShortcutExpander(delegate meta.RESTMapper, client discovery.DiscoveryInterface) shortcutExpander {
+	if client == nil {
+		panic("Please provide discovery client to shortcut expander")
```

Comment:
7 call sites and a quick survey suggests they can all handle an error.  You should be able to update this to return `(shortcutExpander, error)` and avoid the panic here.